### DISCUSSION TITLE
Glasnet settings

### DISCUSSION THREAD

**Discussion by bvk007** [fabianishere/udm-iptv#14] [Show and tell]
Hi Fabian, 

You requested in your documents to add more provider info. Well here is the info for Glasnet (Canal Digitaal)

```IPTV_WAN_INTERFACE="eth4"
IPTV_WAN_RANGES="217.166.224.0/22"
IPTV_WAN_VLAN="37"
IPTV_WAN_DHCP_OPTIONS="-O staticroutes -V IPTV_RG"
IPTV_LAN_INTERFACES="br10" 

```

Cheers,

Bas

**Comment by fabianishere** [comment]
Thank you!

**Comment by DerEnte88** [comment]
Ok, so tried the above configuration on my Glasnet (NL) fiber connection.

Seems to work, but only after deleting a default route created by the iptv container.
There is no difference in including the WAN_DHCP_OPTIONS or not.

This default route also comes back every day. So with every new DHCP lease?

```
# ip route list
default via 10.11.0.1 dev iptv metric 243
etc.
etc.
```

Strange thing is that Glasnet requires a static route over VLAN37 in their documentation.
[iptv.pdf](https://github.com/fabianishere/udm-iptv/files/7636237/iptv.pdf)
However when I manually add this static route IPTV receiver stops working.
With no static route configured everyting works fine.

`# ip route add 185.24.175.0/24 via 10.11.0.1 dev iptv
`

Output of diagnose:
```
# /mnt/data/on_boot.d/15-iptv.sh diagnose
Please share the following output width the developers:
=== Configuration ===
WAN Interface: eth8
WAN Ranges: 217.166.224.0/22
WAN VLAN: 37
LAN Interfaces: br15
=== IP Link and Route ===
43: iptv@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.11.1.184/22 scope global iptv
       valid_lft forever preferred_lft forever
10.11.0.0/22 proto kernel scope link src 10.11.1.184
=== Container Logs ===
udm-iptv: Obtaining IP address for VLAN interface...
udhcpc: started, v1.33.1
udhcpc: sending discover
udhcpc: sending select for 10.11.1.184
udhcpc: lease of 10.11.1.184 obtained, lease time 86400
udm-iptv: NATing IPTV network ranges (if necessary)...
udm-iptv: Setting up igmpproxy..
udm-iptv: Starting igmpproxy..

```

Any thoughts about this? Am I doing something wrong?
If so please tell me, i am no expert :).

Kind Regards,
Koen

**Reply by bvk007** [reply]
Hi Koen,

So, I ran in the same thing...when the container starts the default route is altered and routed over the DHCP address of your IPTV lease instead over the PPPOE connection by glasnet.

My fix was to add a default route and not delete the one added by the container

So do a ```route add default gw <insert your ip here> ppp0```

Cheers,

Bas

**Reply by fabianishere** [reply]
Hi Koen,

Could you try the following DHCP options:
```bash
IPTV_WAN_DHCP_OPTIONS="-o -O subnet -O broadcast -O staticroutes"
```
This should prevent the DHCP client from requesting the default gateway route from the DHCP server (option 3).

**Reply by DerEnte88** [reply]
Hi Fabian,

Was typing my comment below in the meantime. Will try now.

**Reply by DerEnte88** [reply]
@ Fabian,

Yes, Works.
No default gateway is created when running the iptv container now.

**Reply by DerEnte88** [reply]
@fabianishere 
Celebrated to soon. With the above DHCP options IPTV stops working. (Live TV)

**Reply by fabianishere** [reply]
Do you get an IP address from the DHCP server though?

**Reply by fabianishere** [reply]
You could try to add `185.24.175.0/24` to WAN ranges, so that the container can NAT those addresses as well (like described in de guide).

**Reply by DerEnte88** [reply]
> Do you get an IP address from the DHCP server though?

Yes, got the IP address on the iptv interface.
Rebooted my UDMP with the above DHCP options and all works well now. Can't find out why it stopped working.

Added the `185.24.175.0/24` to WAN ranges, seems to have no change (or I haven't found what it's used for).


Many thanks for your help! Great work, happy with it.

**Comment by DerEnte88** [comment]
Hi Bas,
Thanks for the quick response.

_[Content removed, fixed with DHCP options from Fabian one post up.]_


What are your findings regarding the static route that glasnet prescribes?


Koen

**Reply by bvk007** [reply]
I don't use it...completely rely on the igmp proxy by Fabian. However, it seems that it hangs itself after a while, and I haven't figured out why...the proxy restarts itself (I can see it by the DHCP request in the log, and my screen freezes and then resumes)

**Comment by bvk007** [comment]
@DerEnte88 , hi Koen,

How is the stability of your IPTV connection. Mine drops after a while...and I have to restart the container. And also, do you have the NPO2 extra channel, I can see mine, but it doesn't show video. @fabianishere I'm running in debug and verbose mode...I will post something here when I have more output

**Reply by DerEnte88** [reply]
Hey Bas,

Haven't experienced any dropouts, but also didn't watch alot of live TV the past 2 days. What is a while... minutes, hours, days?
I didn't have to restart the container since my last post. So approx up for 48h now.

About NPO2 Extra it's the same for me. Can select the channel but no video\audio.

**Reply by bvk007** [reply]
It's while watching TV, so you'll encounter it while watching.

In the startup script there is a ```--restart on-failure:5``` implemented...so the server will restart 5 times before it goes down.

And the log doesn't show an error...it crashes and just restarts:

```
RECV Leave message      from 192.168.10.25   to 224.0.0.2
RECV Membership query   from 192.168.10.1    to 224.0.252.92
RECV V2 member report   from 10.11.1.90      to 224.0.252.92
The IGMP message was from myself. Ignoring.
The IGMP message was local multicast. Ignoring.
RECV V2 member report   from 192.168.10.25   to 225.0.71.1
Updated route entry for 225.0.71.1 on VIF #0
Adding MFC: 10.11.0.1 -> 225.0.71.1, InpVIf: 1
RECV Membership query   from 192.168.10.1    to 224.0.252.143
Removing MFC: 217.166.226.143 -> 224.0.252.143, InpVIf: 1
udm-iptv: Obtaining IP address for VLAN interface...
udhcpc: started, v1.33.1
udhcpc: sending discover
udhcpc: sending select for 10.11.1.90
udhcpc: lease of 10.11.1.90 obtained, lease time 86400
udm-iptv: NATing IPTV network ranges (if necessary)...
udm-iptv: Setting up igmpproxy..
udm-iptv: Starting igmpproxy..
```

**Reply by fabianishere** [reply]
I have heard this happening when there is a Chromecast on the network.

As a quick fix, you can change it to  `--restart always` to always restart the container on exit.

**Reply by bvk007** [reply]
I've read that too, but I don't do chrome cast...I do have Apple TV's in my network though... I will change that parameter.

**Comment by DerEnte88** [comment]
@bvk007 
Hey Bas,

Can you try and choose "Begin Gemist" from the TV-guide. 
So start the current LiveTV program from the beginning. 

Does not work for me. Replay and On-Demand works properly.

P.s. Still haven't had to restart the container. Seems very stable.

**Reply by DerEnte88** [reply]
Can this be an issue related to the RTSP -> Application Layer Gateway settings found in the infosheet from Glasnet?
There is no default module in the UDM for this as far as my research goes.

**Comment by bvk007** [comment]
Hey Koen, @DerEnte88 

Begin gemist doesn't work for me as well...the other ones do.

Container Is still unstable, I set the restart always parameter. But yesterday during F1 it kept crashing....the container reboots...it even doesn't get an IP lease for a while. Sucks like hell

**Comment by bvk007** [comment]
Checking stability @fabianishere changed the IPTV setup to a Debian package....setup works great btw ;)

**Reply by bvk007** [reply]
3 hours and 45 minutes up....that is a new record, the container setup never managed to be up that long without a restart...

**Comment by DerEnte88** [comment]
Hey Bas @bvk007 

Good to hear your container runs stable. Mine never gave issues during the last month.

Any thoughts about "Begin gemist" and pausing of live TV. Can't seem to get this to work.
Only thing that comes to mind is the RTSP "Application layer Gateway" function in the standard DrayTek routers from Glasnet is used for this.

**Reply by bvk007** [reply]
It isn't a container anymore. Fabian changed the docker (Podman) setup to a Debian package, the igmp proxy now runs as a deamon on the UDM.

About begin gemist and pause live tv...I haven't got that working either...however I didn't spent much time into that. We don't watch that much TV here except for F1, sports and or press conferences for the shitshow we're in right now ;)

I'll try to investigate why this isn't working, It seems like normal internet traffic to me...because "terugkijken" does work. Can you record tv shows btw?

**Reply by bvk007** [reply]
@DerEnte88,

I don't have it working, but RTSP is mandatory for Begin gemist and pauze tv, also some streams that are not working (NPO2 extra). I've fiddled with some firewall settings after googling but no luck. @fabianishere do you happen to know  a solution to get RTSP working on a UDM(pro)?

**Reply by bvk007** [reply]
I've started a new discussion for RTSP
